erpnext/stock/doctype/item/item.py

```diff
@@ -1002,14 +1004,17 @@ def get_last_purchase_details(item_code, doc_name=None, conversion_rate=1.0):
 		order by pr.posting_date desc, pr.posting_time desc, pr.name desc
 		limit 1""", (item_code, cstr(doc_name)), as_dict=1)
 
+	
+	
 	purchase_order_date = getdate(last_purchase_order and last_purchase_order[0].transaction_date
 							   or "1900-01-01")
 	purchase_receipt_date = getdate(last_purchase_receipt and
 								 last_purchase_receipt[0].posting_date or "1900-01-01")
 
-	if (purchase_order_date > purchase_receipt_date) or \
+	if (purchase_order_date >= purchase_receipt_date) or \
```

Comment:
When `last_purchase_order` and `last_purchase_receipt` are both `None`, `purchase_order_date ` and `purchase_receipt_date ` becomes equal and the `>=` messes up tests.

change condition to `if last_purchase_order and (purchase_order_date >= purchase_receipt_date) or \.....`